src/app/slash_plugins/leaderboard.plugin.ts

```diff
@@ -0,0 +1,103 @@
+import { CommandInteraction, User } from 'discord.js';
+import { Command } from '../../common/slash';
+import { IContainer } from '../../common/types';
+import { GameType } from '../../services/gameleaderboard.service';
+
+const gameTypes = ['TicTacToe', 'ConnectFour'];
+
+async function createOpponentPlayerEmbed(
+  container: IContainer,
+  interaction: CommandInteraction,
+  opponent: User,
+  gameEnum: GameType
+) {
+  return container.gameLeaderboardService.createMatchupLeaderboardEmbed(
+    interaction.user,
+    opponent,
+    gameEnum
+  );
+}
+
+function getMatchUpEmbed(
```

Comment:
one of these is marked async, one is not. 
you should just specify the result type to be Promise<MessageEmbed> for both instead